Python on homebrew is busted again

```
  brew unlink python && brew link --overwrite python
  shell: /bin/bash -e {0}
  env:
    RUST_BACKTRACE: 1
    CARGO_INCREMENTAL: false
    SCCACHE_MAX_FRAME_LENGTH: 100000000
    TOOL_DIR: ./tools
    RUSTC_WRAPPER: sccache
    SCCACHE_DIR: /Users/runner/Library/Caches/Mozilla.sccache
    SCCACHE_IDLE_TIMEOUT: 0
    HOMEBREW_NO_INSTALL_FROM_API: 
    HOMEBREW_GITHUB_API_TOKEN: ***
Error: No such keg: /opt/homebrew/Cellar/python
Error: No such keg: /opt/homebrew/Cellar/python
Error: Process completed with exit code 1.
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -499,11 +499,6 @@ jobs:
     steps:
       - uses: Homebrew/actions/setup-homebrew@master
 
-      # Although we don't use Python per se, if Homebrew has an updated version of Python,
-      # the following steps could fail because GitHub Actions mixed up the linkage.
-      - name: Workaround GitHub Actions Python issues
-        run: brew unlink python && brew link --overwrite python
-
       # install postgres with homebrew, but do't let homebrew do an update too.  The update
       # process not only takes forever, but it tends to fail when the internet changes
       #       https://github.com/Homebrew/brew/issues/1670#issuecomment-267096602

